src/Sarif.FunctionalTests/SpecExamples/Comprehensive.sarif

```diff
@@ -74,42 +82,153 @@
                 }
               },
               "fullyQualifiedLogicalName": "collections::list:add"
             }
           ],
           "codeFlows": [
             {
-              "message": "Code flow #1",
+              "message": "Path from declaration to usage",
               "locations": [
                 {
                   "step": 0,
                   "kind": "declaration",
                   "importance": "essential",
-                  "fullyQualifiedLogicalName": "collections::list::add",
+                  "message": "Variable \"p\" declared.",
+                  "snippet": "int *p;",
                   "physicalLocation": {
-                    "uri": "collections/list.cpp",
-                    "uriBaseId": "SRCROOT",
+                    "uri": "file:///home/build/src/collections/list.h",
                     "region": {
-                      "startLine": 12,
-                      "startColumn": 5,
-                      "endLine": 12,
-                      "endColumn": 11,
-                      "offset": 115,
-                      "length": 6
+                      "startLine": 15
                     }
                   },
-                  "snippet": "int n;"
+                  "fullyQualifiedLogicalName": "collections::list:add",
+                  "module": "collections",
+                  "threadId": 52
+                },
+                {
+                  "step": 1,
+                  "kind": "assignment",
+                  "importance": "unimportant",
+                  "snippet": "offset = 0",
+                  "physicalLocation": {
+                    "uri": "file:///home/build/src/collections/list.h",
+                    "region": {
+                      "startLine": 15
+                    }
+                  },
+                  "fullyQualifiedLogicalName": "collections::list:add",
+                  "module": "collections",
+                  "threadId": 52
+                },
+                {
+                  "step": 2,
+                  "kind": "call",
+                  "importance": "essential",
+                  "message": "Uninitialized variable \"p\" passed to method \"add_core\".",
+                  "snippet": "add_core(p, offset, val)",
+                  "callee": "collections::list:add_core",
+                  "physicalLocation": {
+                    "uri": "file:///home/build/src/collections/list.h",
+                    "region": {
+                      "startLine": 25
+                    }
+                  },
+                  "fullyQualifiedLogicalName": "collections::list:add",
+                  "module": "collections",
+                  "threadId": 52
+                }
+              ]
+            }
+          ],
+          "stacks": [
+            {
+              "message": "Call stack resulting from usage of uninitialized variable.",
+              "frames": [
+                {
+                  "message": "Exception thrown",
```

Comment:
Yes, fixed.

---
In reply to: [72281289](https://github.com/Microsoft/sarif-sdk/pull/464#discussion_r72281289) [](ancestors = 72281289)